fn/map_test.go

```diff
@@ -0,0 +1,78 @@
+package fn
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeySet(t *testing.T) {
+	testMap := map[string]int{"a": 1, "b": 2, "c": 3}
+	expected := NewSet([]string{"a", "b", "c"}...)
+
+	result := KeySet(testMap)
+
+	require.True(t, reflect.DeepEqual(result, expected))
```

Comment:
Should use `require.Equal()` instead. This does the same deep equality check but gives an error message with way more useful context than just "should be true" if the check fails.